### ISSUE TITLE
Documentation microsite

### ISSUE THREAD

**Issue by Dhaulagiri** [hashicorp/flight#5]
This will likely need to be decomposed into meta issues, but at a high level we'd like a simple documentation site that offers the following:

- [ ] icon inventory (https://github.com/hashicorp/flight/issues/56)
- [ ] The icon inventory is searchable (https://github.com/hashicorp/flight/issues/55)
- [ ] How to contribute changes (https://github.com/hashicorp/flight/issues/26)
- [ ] Document guidelines around icon styles/conventions and usage (https://github.com/hashicorp/flight/issues/27)
- [ ] A spike on an MVP implementation that will support documenting the above (https://github.com/hashicorp/flight/issues/57)

This will then be folded into our larger documentation effort once that is ready
